pkg/gameserverallocations/find_test.go

```diff
@@ -82,53 +82,6 @@ func TestFindGameServerForAllocationPacked(t *testing.T) {
 				assert.NoError(t, err)
 			},
 		},
-		"one label": {
-			// nolint: dupl
-			list: []agonesv1.GameServer{
-				{ObjectMeta: metav1.ObjectMeta{Name: "gs6", Namespace: defaultNs, Labels: oneLabel, DeletionTimestamp: &n}, Status: agonesv1.GameServerStatus{NodeName: "node1", State: agonesv1.GameServerStateReady}},
-				{ObjectMeta: metav1.ObjectMeta{Name: "gs1", Namespace: defaultNs, Labels: oneLabel}, Status: agonesv1.GameServerStatus{NodeName: "node1", State: agonesv1.GameServerStateReady}},
-				{ObjectMeta: metav1.ObjectMeta{Name: "gs2", Namespace: defaultNs, Labels: oneLabel}, Status: agonesv1.GameServerStatus{NodeName: "node2", State: agonesv1.GameServerStateReady}},
-				{ObjectMeta: metav1.ObjectMeta{Name: "gs3", Namespace: defaultNs, Labels: oneLabel}, Status: agonesv1.GameServerStatus{NodeName: "node1", State: agonesv1.GameServerStateAllocated}},
-				{ObjectMeta: metav1.ObjectMeta{Name: "gs4", Namespace: defaultNs, Labels: oneLabel}, Status: agonesv1.GameServerStatus{NodeName: "node1", State: agonesv1.GameServerStateAllocated}},
-				{ObjectMeta: metav1.ObjectMeta{Name: "gs5", Namespace: defaultNs, Labels: oneLabel}, Status: agonesv1.GameServerStatus{NodeName: "node1", State: agonesv1.GameServerStateError}},
-				{ObjectMeta: metav1.ObjectMeta{Name: "gs6", Namespace: "does-not-apply", Labels: oneLabel}, Status: agonesv1.GameServerStatus{NodeName: "node1", State: agonesv1.GameServerStateReady}},
-			},
-			test: func(t *testing.T, list []*agonesv1.GameServer) {
-				assert.Len(t, list, 3)
-
-				gs, index, err := findGameServerForAllocation(gsa, list)
-				assert.NoError(t, err)
-				if !assert.NotNil(t, gs) {
-					assert.FailNow(t, "gameserver should not be nil")
-				}
-				assert.Equal(t, "node1", gs.Status.NodeName)
-				assert.Equal(t, "gs1", gs.ObjectMeta.Name)
-				assert.Equal(t, gs, list[index])
-				assert.Equal(t, agonesv1.GameServerStateReady, gs.Status.State)
-
-				// mock that the first found game server is allocated
-				list = append(list[:index], list[index+1:]...)
-				assert.Equal(t, agonesv1.GameServerStateReady, list[0].Status.State)
-				assert.Len(t, list, 2)
-
-				gs, index, err = findGameServerForAllocation(gsa, list)
-				assert.NoError(t, err)
-				if !assert.NotNil(t, gs) {
-					assert.FailNow(t, "gameserver should not be nil")
-				}
-				assert.Equal(t, "node2", gs.Status.NodeName)
-				assert.Equal(t, "gs2", gs.ObjectMeta.Name)
-				assert.Equal(t, gs, list[index])
-				assert.Equal(t, agonesv1.GameServerStateReady, gs.Status.State)
-
-				list = nil
-				gs, _, err = findGameServerForAllocation(gsa, list)
-				assert.Error(t, err)
-				assert.Equal(t, ErrNoGameServer, err)
-				assert.Nil(t, gs)
-			},
-			features: fmt.Sprintf("%s=false&%s=false", runtime.FeaturePlayerAllocationFilter, runtime.FeatureStateAllocationFilter),
```

Comment:
yeah this is another one I was probably too quick on, sorry. I think `StateAllocationFilter` needs to be taken out, but you'll have to readjust the test.